include/esp_loader.h

```diff
@@ -148,7 +148,7 @@ esp_loader_error_t esp_loader_flash_start(uint32_t offset, uint32_t image_size,
   *     - ESP_LOADER_ERROR_TIMEOUT Timeout
   *     - ESP_LOADER_ERROR_INVALID_RESPONSE Internal error
   */
-esp_loader_error_t esp_loader_flash_write(const void *payload, uint32_t size);
+esp_loader_error_t esp_loader_flash_write(void *payload, uint32_t size);
```

Comment:
Let's not remove const here, as mentioned, payload does not change.

Reply:
`Payload` (a pointer to the buffer) is aliased by `data`, and padding is applied to that same buffer via the `data` pointer, which only exists for purpose of casting `void*` to `uint8_t*`. Removing const-ness is an error. What am I missing?

Reply:
Exactly. Keep the const in the declaration and don't cast it away in the function implementation.

Reply:
> Keep the const in the declaration and don't cast it away in the function implementation.

But `esp_loader_flash_write` _does_ in fact modify the payload in:
```c++
    while (padding_bytes--) {
        data[padding_index++] = PADDING_PATTERN;
    }
```

So there are two options:

1. Keep `const void *payload` in declaration, `const void *payload` in the implementation, and modify `esp_loader_flash_write` to _not_ do any padding and require the caller to do it. (May be a breaking change.)
2. Keep the existing behavior (`esp_loader_flash_write` performs the padding), then remove the `const` in the declaration, as done here.

I think 2) makes sense in a bugfix release, 1) could be considered as a breaking change in the next major release.

Reply:
My bad, i completely missed that part, thanks @igrr . Let's keep it as it is.